src/libexpr/primops/context.cc

```diff
@@ -67,10 +61,51 @@ static void prim_unsafeDiscardOutputDependency(EvalState & state, const PosIdx p
 static RegisterPrimOp primop_unsafeDiscardOutputDependency({
     .name = "__unsafeDiscardOutputDependency",
     .arity = 1,
+    .doc = R"(
+      TODO! Old comment is not good.
+
+      Sometimes we want to pass a derivation path (i.e. `pkg.drvPath`) to a builder without causing the derivation to be built
+      (for instance, in the derivation that builds NARs in nix-push, when doing source-only deployment).
+      This primop marks the string context so that `builtins.derivation` adds the path to `drv.inputSrcs` rather than `drv.inputDrvs`.
```

Comment:
Needs better docs!

Reply:
Don't hesitate ;)

Reply:
On this line specifically I think `drv` is undefined, and `inputSrcs` and `inputDrvs` should be links to definitions, which we don't really have.
The only mention is in `nix derivation show`, which is a questionable place to document these two concepts.